### PR TITLE
Add WebAssembly features as a new category

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adds a brand new category for WebAssembly features -- that is, features within the WebAssembly code itself rather than just the JavaScript API.  The results were collected by the mdn-bcd-collector project (v10.0.0).

CC @MendyBerger

### CODE REVIEW

**Review by Elchi3 — COMMENTED** [review]
Some questions:
-  Unlike other top-level categories, the webassembly features are added flat without any folders or categories. Will this scale? 
- Are there spec_urls for these features? It says standards_track true, so I guess there must be.
- Are there mdn_urls for these features?
- Will this PR mean a major version bump for BCD given a new top-level tree is added?

**Comment by queengooborg** [comment]
> * Unlike other top-level categories, the webassembly features are added flat without any folders or categories. Will this scale?

When @MendyBerger wrote the PR to add this category into the collector, they used `webassembly.features.*` instead of just `webassembly.*`, but I later thought a subfolder was redundant.  Happy to add it back though!

> * Are there spec_urls for these features? It says standards_track true, so I guess there must be.

@MendyBerger, you know a lot more about WebAssembly than we do -- do you know if there are spec URLs associated with these features?

> * Are there mdn_urls for these features?

Not yet as far as I know, but I believe there are plans to write them!

> * Will this PR mean a major version bump for BCD given a new top-level tree is added?

Not a major version, but a minor version, since it's adding a new backwards-compatible feature!

**Comment by MendyBerger** [comment]
Subfolder:
I don't expect too many features on the wasm side of things. It's pretty low level, so development of new features are probably going to slow down over time. But I'm not working on the specs so take my opinion with a grain of salt.
I don't think that a subfolder would make a huge difference either way. Whatever works I guess :shrug: .

Spec urls:
There sure are, I can add them in. 
I'm assuming they would go in the wasm.json file in collector, right?

**Comment by Elchi3** [comment]
Thanks for your comments!

- no sub folders then
- minor version bump
- let's add spec_urls before merging this.

**Comment by MendyBerger** [comment]
@queengooborg see https://github.com/queengooborg/browser-compat-data/pull/53

**Comment by queengooborg** [comment]
Got the spec URLs added, and made the appropriate changes to typedefs, the readme and more -- this is now ready to merge!

File: test/linter/test-obsolete.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ const categoriesToCheck = [
   // 'javascript',
   // 'mathml',
   // 'svg',
+  // 'webassembly',
```
Comment (Elchi3) [inline test/linter/test-obsolete.ts:24]: Can we enable this?

File: test/linter/test-spec-urls.ts [hunk 2]
```diff
@@ -38,6 +38,13 @@ const specsExceptions = [
   // Remove once https://github.com/whatwg/html/pull/8502
   // is merged and the new URL is live
   'https://wicg.github.io/navigation-api',
+
+  // Proposals for WebAssembly
+  'https://github.com/WebAssembly/spec/blob/master/proposals',
```
Comment (Elchi3) [inline test/linter/test-spec-urls.ts:43]: master? Isn't main the default branch for these?

**Review by Elchi3 — APPROVED** [review]
Thanks, let's do this then 👍

**Pull request merged**